nomos-services/membership/src/backends/mock.rs

```diff
@@ -70,6 +70,12 @@ impl MembershipBackend for MockMembershipBackend {
     ) -> Result<HashMap<ServiceType, MembershipProviders>, MembershipBackendError> {
         let block_number = update.block_number;
 
+        if block_number <= self.latest_block_number {
+            return Err(MembershipBackendError::Other(
```

Comment:
This case deserves it's own error I think, like `MembershipBackendError::BlockFromPast`. This change could also introduce `thiserror` to the MembershipBackendError where we could add the description. I would keep `Other` variant for db, io or implementation specific errors.

Reply:
It depends what we want to do in this case, but I agree this is the most flexible way. Will add the suggestion, thanks

Reply:
https://github.com/logos-co/nomos/pull/1333/commits/0865970c8bece87a21d7483d1cb38820aa872ace. @bacv Let me know if this is better